src/anim.ts

```diff
@@ -139,14 +139,8 @@ function step(state: State): void {
 
 function animate<A>(mutation: Mutation<A>, state: State): A {
   // clone state before mutating it
-  const prevPieces: cg.Pieces = {};
-  // clone pieces
-  for (let key in state.pieces) {
-    prevPieces[key] = {
-      role: state.pieces[key].role,
-      color: state.pieces[key].color
-    };
-  }
+  const prevPieces: cg.Pieces = {...state.pieces};
```

Comment:
yeah this is a shallow copy, it didn't look we actually changing the pieces anymore